### ISSUE TITLE
[vagrant provider] vbox provider should be named virtualbox -> review load_config

### ISSUE THREAD

**Issue by msimonin** [BeyondTheClouds/enos#111]
To reproduce : 

```
provider: 
  type: vagrant
  option: vbox
```
Leads to 
```
The provider 'vbox' could not be found, but was requested to
back the machine 'enos-127'. Please use a provider that exists.
```

Note that `provider:vagrant` works. In this case `vagrant` is called without any env variables and default to virtualbox.

**Comment by msimonin** [comment]
I think I will review how provider options are loaded. 
Each provider should be able to specify the default values for its provider section. 
For example for vagrant this could be : 
```
DEFAULT_PROVIDER_CONFIG = {
    'backend': 'virtualbox'
}
```

For g5k : 

```
DEFAULT_PROVIDER_CONFIG = {
    'name': 'kolla-discovery',
    'site': 'rennes',
    'walltime': '02:00:00',
    'env_name': 'jessie-x64-min',
    'reservation': None,
    'vlans': {rennes: "{type='kavlan'}/vlan=1"},
    'role_distribution': ROLE_DISTRIBUTION_MODE_STRICT,
    'single_interface': False,
    'user': 'root'
}
```
In the latter, I propose to nest all g5k related option inside provider.

**Comment by msimonin** [comment]
addressed by #112
